src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    if (year < 30) {
+      dateParts['YYYY'] = `20${dateParts['YY']}`;
+    } else {
+      dateParts['YYYY'] = `19${dateParts['YY']}`;
+    }
+  }
+
+  if (dateParts['YYYY'] && !dateParts['YY'] && toFormat.includes('YY')) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
```

Comment:
Similarly, the logic for converting 'YYYY' to 'YY' could also be refactored into a separate function. This will not only make the `formatDate` function cleaner but will also allow for easier testing and potential reuse of the conversion logic.